orangecontrib/spectroscopy/utils/Peak_line.py

```diff
@@ -0,0 +1,569 @@
+# -*- coding: utf-8 -*-
+from AnyQt import QtGui, QtCore
+from pyqtgraph import Point
+from pyqtgraph import GraphicsObject
+from pyqtgraph import TextItem
+from pyqtgraph import ViewBox
+from pyqtgraph import functions as fn
+import numpy as np
+
+__all__ = ['PeakLine', 'PeakLabel']
+
+"""
+    A modification of the Infinite Line Graphicsobject.
+    Uses the base function as a means to create labels specific for peak
+    marking in spectra. 
+"""
+
+
+class PeakLine(GraphicsObject):
+    """
+    **Bases:** :class:`GraphicsObject <pyqtgraph.GraphicsObject>`
```

Comment:
For subclassing or encapsulation of InfiniteLine see Orange.widgets.evaluate.owrocanalysis.InfiniteLine (subclassing with small changes) or orangecontrib.spectroscopy.widgets.gui.MovableVLine (for encapsulation with friendlier signals). Is there any particular reason why you did not use MovableVLine (or its slight modification) for your PR?